components/esp-tls/esp_tls.c

```diff
@@ -123,7 +123,7 @@ static ssize_t tcp_read(esp_tls_t *tls, char *data, size_t datalen)
 
 static ssize_t tcp_write(esp_tls_t *tls, const char *data, size_t datalen)
 {
-    return send(tls->sockfd, data, datalen, 0);
+    return send(tls->sockfd, data, datalen, MSG_MORE);
```

Comment:
I don't think we could hard-code this flag here. It makes sense for this specific use-case, but it's not universally the case for other usages. In fact, users usually prefer that the data are sent immediately, even for TCP, so we typically suggest that people set `TCP_NODELAY` to disable naggling.

I'd suggest adding this to `esp_tls` config options, something like:
```patch
@@ -200,6 +200,7 @@ typedef struct esp_tls_cfg {
     esp_tls_addr_family_t addr_family;      /*!< The address family to use when connecting to a host. */
     const int *ciphersuites_list;           /*!< Pointer to a zero-terminated array of IANA identifiers of TLS ciphersuites.
                                                 Please check the list validity by esp_tls_get_ciphersuites_list() API */
+    int socket_flags;
 } esp_tls_cfg_t;
```